torch/distributed/distributed_c10d.py

```diff
@@ -1389,7 +1414,7 @@ def all_gather_object(object_list, obj, group=group.WORLD):
     input_tensor, local_size = _object_to_tensor(obj)
     group_backend = get_backend(group)
     is_nccl_backend = group_backend == Backend.NCCL
-    current_device = torch.device("cpu")
+    current_device: Union[int, torch.device] = torch.device("cpu")
```

Comment:
torch.cuda.current_device() returns int, and torch.device() returns torch.device.